app/models/goal.py

```diff
@@ -1,6 +1,38 @@
 from flask import current_app
+from sqlalchemy.orm import backref
 from app import db
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    # genres = db.relationship("Genre", secondary="books_genres", backref="books")
+    tasks = db.relationship("Task", backref="goal", lazy=True)
+
+    def to_dict(self):
+        if self.list_of_task_ids():
```

Comment:
Storing this result of function call here in a variable (ie `task_ids = self.list_of_task_ids()`) and then testing could create a slight optimization, because then the variable could be used on line 19 (`"task_ids": task_ids`) instead of calling the function again.